runtime/compiler/runtime/SymbolValidationManager.hpp

```diff
@@ -0,0 +1,1337 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef SYMBOL_VALIDATION_MANAGER_INCL
+#define SYMBOL_VALIDATION_MANAGER_INCL
+
+#include <algorithm>                       // for std::max, etc
+#include <map>
+#include <set>
+#include <stddef.h>                        // for NULL
+#include <stdint.h>                        // for int32_t, uint8_t, etc
+#include "env/jittypes.h"                  // for uintptrj_t
+#include "j9.h"
+#include "j9nonbuilder.h"
+#include "infra/TRlist.hpp"
+#include "env/TRMemory.hpp"
+#include "env/VMJ9.h"
+#include "runtime/Runtime.hpp"
+
+namespace TR {
+
+struct SymbolValidationRecord
```

Comment:
are these types needed outside of SymbolValidationManager? If not, I'd suggest declaring them inside SymbolValidationManager.cpp or at least non-public inside SymbolValidationManager